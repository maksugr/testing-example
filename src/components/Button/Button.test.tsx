import { describe, it, expect, vi } from "vitest";
import { render, screen, fireEvent } from "@testing-library/react";

import { Button } from ".";

describe("Button component", () => {
    it("renders with the correct label", () => {
        const onClick = vi.fn();
        render(<Button label="Click Me" onClick={onClick} />);

        const button = screen.getByTestId("button");
        expect(button).toBeInTheDocument();
        expect(button.textContent).toBe("Click Me");
    });

    it("calls onClick when clicked", () => {
        const onClick = vi.fn();
        render(<Button label="Click Me" onClick={onClick} />);

        const button = screen.getByTestId("button");
        fireEvent.click(button);
        expect(onClick).toHaveBeenCalledTimes(1);
    });

    it("applies disabled state correctly", () => {
        const onClick = vi.fn();
        render(<Button label="Click Me" onClick={onClick} disabled={true} />);

        const button = screen.getByTestId("button");
        expect(button).toBeDisabled();

        fireEvent.click(button);
        expect(onClick).not.toHaveBeenCalled();
    });
});

describe("Button component snapshot", () => {
    it("matches default snapshot", () => {
        const { container } = render(
            <Button label="Click me" onClick={() => {}} />
        );
        expect(container).toMatchSnapshot();
    });

    it("matches disabled snapshot", () => {
        const { container } = render(
            <Button label="Can't click" onClick={() => {}} disabled />
        );
        expect(container).toMatchSnapshot();
    });
});
