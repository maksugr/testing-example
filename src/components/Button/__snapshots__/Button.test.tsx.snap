// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button component snapshot > matches default snapshot 1`] = `
<div>
  <button
    data-testid="button"
  >
    Click me
  </button>
</div>
`;

exports[`Button component snapshot > matches disabled snapshot 1`] = `
<div>
  <button
    data-testid="button"
    disabled=""
  >
    Can't click
  </button>
</div>
`;
